proposals/lip-0055.md

```diff
@@ -105,16 +115,42 @@ The full processing of a block is organized as follows.
 5. **Assets verification**: Each module verifies the respective entry in the block assets. If any check fails, the block is discarded and has no further effect.
 6. **Block forwarding**: After the initial checks, the full block is forwarded to a subset of peers.
 7. **Before transactions execution**: Each module can define protocol logic that is executed *before* the transactions contained in the block are processed.
-8. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If the verification fails, the transaction is invalid and the whole block is discarded.
-9. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified.
-10. **Before command execution**: Each module can define protocol logic that is processed *before* the command has been executed.
-11. **Command execution**: The command corresponding to the `moduleID`-`commandID` combination is executed.
-12. **After command execution**: Each module can define protocol logic that is processed *after* the command has been executed.
+8. **Transaction verification**: Each module can define protocol logic that verifies a transaction, possibly by accessing the state store. If an error occurs, the transaction is invalid and the whole block is discarded.
+9. **Command verification**: The command corresponding to the `moduleID`-`commandID` combination is verified. If an error occurs, the transaction is invalid and it is not included in the block.
```

Comment:
Here and below, I think it would be better to write the following similar to step 8:

> If an error occurs, the transaction is invalid and the whole block is discarded.

The section here is about the block processing, not the block generation, so transactions cannot be removed from a block.

Reply:
fixed